packages/calcite-components/src/components/radio-button/radio-button.tsx

```diff
@@ -425,21 +423,32 @@ export class RadioButton
       case "ArrowLeft":
       case "ArrowUp":
         event.preventDefault();
-        this.selectItem(
-          radioButtons,
-          getRoundRobinIndex(Math.max(currentIndex - 1, -1), radioButtonsLength),
-        );
+        this.selectItem(radioButtons, this.getNextNonDisabledIdx(radioButtons, currentIndex, -1));
         return;
       case "ArrowRight":
       case "ArrowDown":
         event.preventDefault();
-        this.selectItem(radioButtons, getRoundRobinIndex(currentIndex + 1, radioButtonsLength));
+        this.selectItem(radioButtons, this.getNextNonDisabledIdx(radioButtons, currentIndex));
         return;
       default:
         return;
     }
   }
 
+  private getNextNonDisabledIdx(
+    radioButtons: RadioButton["el"][],
+    startIdx: number,
+    dir = 1,
```

Comment:
For maintainability, let’s use semantic values here. From an API perspective, it’s unclear what 1 represents as a direction.